frontend/src/wsApi.ts

```diff
@@ -1,19 +1,32 @@
-import { useState, useRef, useEffect } from "react";
+import { useState, useRef, useEffect, MutableRefObject } from "react";
 import { config } from "./Constants";
+import { LetterType } from "./Grid";
+import { DrawingEvent } from "./Crossword";
 
-export const useWsApi = (url) => {
-  const socket = useRef(null);
-  const onExternalDraw = useRef(() => {
-    console.error("No draw function set");
-  });
-  const onExternalWrite = useRef(() => {
-    console.error("No write function set");
-  });
-  const [readyState, setReadyState] = useState("CONNECTING");
+type EventTypes = "DRAWING_EVENTS" | "DRAWING_HISTORY" | "WRITE_HISTORY"
+
+interface DataInterface {
```

Comment:
No need to have `Interface` in the name IMO, `WebSocketData`?

Reply:
As long as it does not create any mixups with other functions, variables etc.